src/main/java/com/sofrelgroup14/ci/controller/BuildController.java

```diff
@@ -1,34 +1,130 @@
+    1) https://stackabuse.com/get-http-post-body-in-spring/
+    2) https://www.baeldung.com/spring-request-response-body
+    3) https://stackoverflow.com/a/6019761
+    */
+    /**
+     * Builds the branch specified in "ref" attribute of the webhook payload
+     * (this is the branch that was pushed to), runs tests on code in this branch,
+     * and finally logs the output to the database.
+     *
+     * Does this by cloning the repo, checking out the branch, building
+     * and running tests using 'mvn' command, and then deletes the repo.
+     *
+     * Example payload (irrelevant parts omitted):
+     * {
+     *  ...
+     *  "ref": "refs/heads/webhooks-playground",
+     *  "before": "c130af7052edd8cc063dd28df6441262cb782822",
+     *  "after": "bee5e5bd2e8da4bcde79f8a0483bda650bdb1b95",
+     *  ...
+     * }
+     * would run:
+     * 'git clone -b webhooks-playground <repo-url> repo'
+     * 'mvn verify --file repo/pom.xml'
+     * ...save the output to the database...
+     * 'rm -Rf repo'
+     *
+     * What HTTP-payload looks like can be seen in Settings -> Webhooks
+     * on GitHub (click an active webhook and scroll down).
+     *
+     * @param body The HTTP POST request body.
+     */
     @RequestMapping(value = "", method = RequestMethod.POST)
-    public Logs buildAndCreateLog() {
-        Logs log = new Logs(ObjectId.get(), true, "great job", Instant.now());
-        repository.save(log);
-        return log;
+    public void buildAndCreateLog(@RequestBody String body) {
 
-    }
+        // TODO (notification): Set commit status to 'pending'
+
+        String ref;
+        String commitHash = ""; // Remove error: "commitHash may not have been initialized"
+        String branchName = ""; // -||-
+
+        // Parse the JSON in the POST body to extract commit-hash and branch name.
+        // (Source for JSON parsing strings automatically: https://stackoverflow.com/a/5245881)
+        try {
+            JSONObject jsonObject = new JSONObject(body);
+
+            ref = jsonObject.getString("ref"); // Will be "refs/heads/branch-name"
+            String[] tmp = ref.split("/");
+            branchName = tmp[tmp.length-1];
+
+            commitHash = jsonObject.getString("after");
+
+        } catch (JSONException e){
+            e.printStackTrace();
+        }
 
+        // Clone repo, run mvn verify, post output to db, remove repo
+        // ProcessBuilder:s have to be wrapped in try-catch block.
+        try {
+            ProcessBuilder pbClone = new ProcessBuilder(
+              "git", "clone", "-b", branchName, "https://github.com/sofrel-group14/ci.git", "repo"); // Could be useful to save time and space: '--single-branch' ("Clone only the history leading to the tip of a single branch, either specified by the --branch option or the primary branch remote's HEAD points at.")
+            pbClone.start().waitFor(); // Start process and block this program thread until process has finished.
```

Comment:
For some reason `ProcessBuilder`s need each command to be in a separate argument (i.e. you cannot do `"git clone -b", branchName, ...`).
It looks pretty ugly (I think) but at least now you know why that is